src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -115,7 +115,9 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 				if(newPhysicalLevel == oldPhysicalLevel && newPhysicalLevel == PHYSICAL_THREAT_LEVEL.HIGH) {
 					String password = request.getPartAsStringFailsafe("masterPassword", MAX_PASSWORD_LENGTH);
 					String oldPassword = request.getPartAsStringFailsafe("oldPassword", MAX_PASSWORD_LENGTH);
-					if(password != null && oldPassword != null && password.length() > 0 && oldPassword.length() > 0) {
+					String confirmPassword = request.getPartAsStringFailsafe("confirmMasterPassword", MAX_PASSWORD_LENGTH);
+					boolean passwordsMatch = password.equals(confirmPassword);
```

Comment:
According to the documentation provided it does return an empty string if the "part" does not exist, it did probably return null sometime in the past but that no longer holds true.